### ISSUE TITLE
Filtering functions don't work

### ISSUE THREAD

**Issue by alexarje** [fourMs/MGT-python#274]
I just tried the various filtering functions, threshold and type, but they don't seem to work. See here for a threshold of 0.0: 

![Screenshot from 2023-01-12 09-33-49](https://user-images.githubusercontent.com/114316/212017301-ec80b964-c2a0-498f-b617-5daf52881292.png)

Is this related to the FFmpeg change?

**Comment by joachimpoutaraud** [comment]
I tested the function on both Windows and Linux OS but it seems to work properly. Did you use the last version of musicalgestures (1.2.6)?
 
![test](https://user-images.githubusercontent.com/63192177/212036489-2a69e393-56c4-4f8b-a064-246bcda0cbe8.png)

Note that the default non-thresholded motiongram using `ffmpeg` is a bit more noisy compared to using `OpenCV`. This is due to the different kind of processing involved as well as the temporal averaging denoiser that is applied (see #109)

**Comment by alexarje** [comment]
Yes, I get some effect, but perhaps less than I had expected. The filter goes from 0 to 1 right? So I would expect that when you filter at 0.0 you would get a lot of noise, while at 0.9 very little signal?

**Comment by joachimpoutaraud** [comment]
Yes from 0 to 1, the problem is related to the way FFmpeg filters each motion frame. I tried to understand what was the difference with filtering without using FFmpeg but it is still difficult to get an answer. If you want I can add the possibility to use traditional filtering but it will take more time to process.

Here are some examples with a `threshold` set to zero:
**FFMPEG >>> Processing time: 16.667 s** 
![dance_mgy_0](https://user-images.githubusercontent.com/63192177/215770766-783db5c9-768b-4dea-839f-b463d32c5844.png)

**TRADITIONAL >>> Processing time: 81.574 s**
![dance_mgy](https://user-images.githubusercontent.com/63192177/215770889-8760939a-e686-4aa3-89df-65b70f5f5dc8.png)

**Comment by alexarje** [comment]
I test this yesterday, and see that the visual differences are quite large between the old and new function. See motiongrams of the same video made with the default settings of the old and new.

![gJB_sFM_c09_d08_mJB0_ch08_mgy-1-684x1024](https://github.com/fourMs/MGT-python/assets/114316/802a6150-7122-4465-8a7f-bb8005092875)

![aist_mgx](https://github.com/fourMs/MGT-python/assets/114316/eab83342-1135-4cf3-b50a-d4c1cc569855)

It would be good to investigate a little more what we can do to improve the filtering of the motiongrams with the new function. The speed benefit is huge, so it would be very nice to work in FFmpeg. Perhaps a midterm solution could be to add the option of using both?

**Comment by alexarje** [comment]
Fixed in **v1.4.2**.

The root cause was a corrupted FFmpeg filter name: a botched `thresh`→`threshold` rename had produced `thresholdold` in `_filter.py`, so the FFmpeg `threshold` filter was never applied (and motion rendering silently produced no frames). The `threshold` and `filtertype` options in `motiongrams()` / `motion()` now work as expected.

Verified: with the fix, different thresholds produce different output (e.g. `threshold=0.0` → mean pixel value ~13.2 vs `threshold=0.5` → ~4.5 on the example video).

Available now via `pip install --upgrade musicalgestures` (>= 1.4.2).
